packages/replayio/src/utils/spawnProcess.ts

```diff
@@ -29,30 +34,28 @@ export function spawnProcess(
     // github.com/replayio/replay-cli/pull/344#discussion_r1553258356
     spawned.unref();
   } else {
+    const stderr = collectData(spawned.stderr);
+
     spawned.on("error", error => {
       deferred.rejectIfPending(error);
-
-      if (onError) {
-        onError(error);
-      }
     });
 
     spawned.on("spawn", () => {
-      if (onSpawn) {
-        onSpawn();
-      }
+      onSpawn?.();
     });
 
     spawned.on("exit", (code, signal) => {
-      if (code || signal) {
```

Comment:
According to the node docs we were basically always rejecting this deferred here:
> The 'exit' event is emitted after the child process ends. If the process exited, code is the final exit code of the process, otherwise null. If the process terminated due to receipt of a signal, signal is the string name of the signal, otherwise null. One of the two will always be non-null.

I wonder if that's the reason why we have the code that silences the uncaught promises (this one was always uncaught!)

Reply:
> I wonder if that's the reason why we have the code that silences the uncaught promises (this one was always uncaught!)

No. We were calling "rejectIfPending" and "resolveIfPending" which are no-ops if the deferred is not pending.

The reason we had _that_ code was in case you CTRL+C while a _prompt_ (like the Y/n prompts or select-recordings list) was active (bc those are promises)

Reply:
Right, but this deferred was pending at the time when this got called. 

So the scenario I'm thinking about goes like this:
1. we got ENTER to confirm the prompt
2. that killed the browser process
3. since it got killed we got `signal` here and we called `deferred.rejectIfPending`
4. nothing was `await`ing the `deferred` returned from this function
5. that could result in an unhandled rejection

Note that I really didn't test it out but the steps above do check out logically

Reply:
Hmm, I just tested that theory (by removing the conditional in `uncaughtException`) and it didn't print anything so I dunno.

Reply:
I couldn't make an actual unhandledrejection error logged into the console - but after removing `process.on("uncaughtException", ...)` handler and a noop `.catch` in `createDeferred` I managed to make the process exit (with an error logged into the console).

To trigger this situation in combination with `spawnProcess` I had to force this deferred to reject on exit... cause even when it's `.kill`ed we still get `code === 0` from Replay Chromium (which doesn't match my expectation cause it should re-raise the received signal or something like that).

Anyway, I'm just posting this as interesting trivia and nothing more right now 😉